docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -1370,6 +1370,9 @@ userland:
 -  ``Query::HINT_CUSTOM_TREE_WALKERS`` - An array of additional
    ``Doctrine\ORM\Query\TreeWalker`` instances that are attached to
    the DQL query parsing process.
+-  ``Query::HINT_INCLUDE_META_COLUMNS`` - A boolean that causes 
+   meta columns like foreign keys and discriminator columns to be 
+   selected and returned as part of the query result.
```

Comment:
If it's not really usable, then we should maybe not document it at all, or at least let the users who are curious about it know that it's used internally. Like you, they might be able to leverage it through a few hacks, but it's clearly not designed to be used outside Doctrine itself. That could be challenged, and the design improved, I suppose.